integration-templates/sharepoint-online/syncs/file-sync.ts

```diff
@@ -0,0 +1,125 @@
+import type { NangoSync, FileMetadata } from '../../models';
+import type { PaginationParams, PaginationResponse } from '../helper/pagination';
+import paginate from '../helper/pagination.js';
+import type { SharePointMetadata } from '../types';
+
+export default async function fetchData(nango: NangoSync): Promise<void> {
+    const metadata = (await nango.getMetadata()) as SharePointMetadata;
+
+    if (!metadata || !metadata.sitesToSync || metadata.sitesToSync.length === 0) {
```

Comment:
Can you add `Array.isArray(metadata.sitesToSync)` please? ☺️

Reply:
Thanks for pointing that out! 😊 I've added this to ensure `sitesToSync` is properly initialized as an array.